Refactor generating shopping lists:

PDELIMATA opened the pull request:
- creating new shopping list
- adding recipes to shopping list
- view of all shopping list of user
- generate shopping list based on a products in fridge and recipes added to shopping list
- deleting shopping list

PDELIMATA committed:
Refactor generating shopping lists:
- creating new shopping list
- adding recipes to shopping list
- view of all shopping list of user
- generate shopping list based on a products in fridge and recipes added to shopping list
- deleting shopping list

PDELIMATA commented:
Dobra to tak, troche tutaj nazmieniałem więc może się okazać że musicie dropnąć bazę żeby na nowo ją postawiło.

Nie miałem pomysłu jak to inaczej zrobić więc musiały się pojawić dwa @Transactionale w ShoppingListRepository - probowałem je dać w serwisie nad odpowiednimi metodami ale nie działało, a tak wydaje się wszystko pracować poprawnie. 

Wiem że brakuje testów (dodałem osobny task z tym) oraz że niektóre części kodu możnaby zoptymalizować.

PDELIMATA committed:
Merge branch 'develop' into JJDZR6JTH-74

szablewski commented on src/main/java/com/infoshareacademy/controller/RecipeController.java:
```diff
@@ -103,4 +108,12 @@ public String deleteRecipe(@PathVariable Long id) {
+
+    @GetMapping("/{recipeId}/shoppinglist/{shoppingListId}")
+    public String addRecipeToShoppingList(@PathVariable Long recipeId, @PathVariable Long shoppingListId) {
+        shoppingListService.addRecipeToShoppingList(recipeService.getRecipeById(recipeId),shoppingListId);
```
Brakuje mi komunikatów o dadniu produku do listy

szablewski commented on src/main/java/com/infoshareacademy/controller/ShoppingListController.java:
```diff
@@ -17,7 +24,34 @@ public ShoppingListController(ShoppingListService shoppingListService) {
+        model.addAttribute("getShopping", shoppingListService.findAllShoppingLists());
         return "shoppinglist";
     }
+    @GetMapping("/shoppinglist/new")
```
```suggestion
    @GetMapping("/shopping-list")
```

PDELIMATA replied:
Ale juz mamy jeden @GetMapping na endpointcie /shoppinglist. To się chyba będzie gryzło?

szablewski commented on src/main/java/com/infoshareacademy/controller/ShoppingListController.java:
```diff
@@ -17,7 +24,34 @@ public ShoppingListController(ShoppingListService shoppingListService) {
+    @GetMapping("/shoppinglist/new")
+    public String createShoppingListForm(Model model) {
+        model.addAttribute("shoppinglist", new ShoppingList());
+        return "create-shoppinglist";
```
```suggestion
        return "create-shopping-list";
```

szablewski commented on src/main/java/com/infoshareacademy/controller/ShoppingListController.java:
```diff
@@ -17,7 +24,34 @@ public ShoppingListController(ShoppingListService shoppingListService) {
+        }
+        shoppingListService.saveShoppingList(shoppingList);
+        return "redirect:/shoppinglist";
+    }
```
Brak komunikatu o utworzeniu listy zakupów

szablewski commented on src/main/java/com/infoshareacademy/controller/ShoppingListController.java:
```diff
@@ -17,7 +24,34 @@ public ShoppingListController(ShoppingListService shoppingListService) {
+    @GetMapping("/shoppinglist/list/{shoppingListId}")
+    public String deleteRecipe(@PathVariable Long shoppingListId) {
+        shoppingListService.deleteShoppingListById(shoppingListId);
+        return "redirect:/shoppinglist";
```
brak komunikatu o usunieciu listy zakupów

szablewski commented on src/main/resources/templates/oneshoppinglist.html:
```diff
@@ -0,0 +1,60 @@
+<!DOCTYPE html>
```
zalecam rozbicie taki zlepek "oneshoppinglist" na "details-shopping-list"

szablewski commented on src/main/resources/templates/shoppinglist.html:
```diff
@@ -29,8 +29,34 @@
+    </table>
 
-    <table class="table table-striped table-bordered">
+<!--    <table class="table table-striped table-bordered">
```
clean code - nie wrzucamy do repozytorium zakomentowanego kodu

szablewski reviewed (commented):
Kilka podstawowych błędów przy znawenictwie endpointów. 
Metoda compareListOfRecipeAndProductsFromFridge w shoppinglistyservice do rozbicia na mniejsze części.
Nazwy niektórych widoków należy rozdzielić "-" aby były czytelniejsze
W całym projekcie brakuje powiadomień o wykonanym działaniu. 
W całym projekcie brakuje nawet podstawowej obsługi błędów tzn łapania wyjątków.
W całym projekcie brakuje logowania tzn loggingu.

PDELIMATA committed:
Fix endpoints names.
Fix methods in ShoppingListService.
Add popups and loggers when some action happend.

PDELIMATA committed:
Merge remote-tracking branch 'origin/JJDZR6JTH-74' into JJDZR6JTH-74

# Conflicts:
#	pom.xml

PDELIMATA committed:
Fix endpoints names.
Fix methods in ShoppingListService.
Add popups and loggers when some action happend.

m-klimek committed:
Merge branch 'develop' into JJDZR6JTH-74
